agent/comms.go

```diff
@@ -81,8 +81,8 @@ func (a *orbAgent) startComms(config config.MQTTConfig) error {
 	var err error
 	a.client, err = a.connect(config)
 	if err != nil {
-		a.logger.Error("connection failed", zap.Error(err))
-		return err
+		a.logger.Error("connection failed", zap.String("key", config.Key), zap.String("channel", config.ChannelID), zap.String("id", config.Id), zap.Error(err))
```

Comment:
we can't log the key, it's essentially a password. and let's use "agent_id" and never just "id" because it's hard to search for that later, or understand which id it is